app/src/main/java/com/openclassrooms/realestatemanager/repository/EstateRepository.kt

```diff
@@ -18,6 +18,20 @@ class EstateRepository(estateApi: EstateApi) {
         estateListLiveData.value = mutableEstateList.toList()
     }
 
+    fun editEstate(editedEstate: Estate) {
+        val estateList: List<Estate> = estateListLiveData.value ?: return
+        val mutableEstateListWithEditedEstate: MutableList<Estate> = mutableListOf()
+        for (estate in estateList) {
+            if (estate.id == editedEstate.id) {
+                mutableEstateListWithEditedEstate.add(editedEstate)
+            } else {
+                mutableEstateListWithEditedEstate.add(estate)
+            }
+        }
+        estateListLiveData.value = mutableEstateListWithEditedEstate.toList()
```

Comment:
```suggestion
        estateListLiveData.value = estateList.map { if (it.id == editedEstate.id) editedEstate else it } 
```

https://kotlinlang.org/docs/collection-transformations.html pour la syntaxe et d'autres exemples d'opérateurs utiles sur les listes en Kotlin :)

Reply:
https://github.com/Tgarfr/Project-9-RealEstateManager/pull/6/commits/08df7b74e366e5d541807f4980f4bb38f2eae45b